src/contrib/mlir/compiler.cpp

```diff
@@ -187,7 +187,7 @@ mlir::Type MLIRCompiler::get_mlir_type(const descriptor::Tensor* tensor)
 // Converts an nGraph element type into an MLIR type.
 mlir::Type MLIRCompiler::get_mlir_type(const element::Type& type)
 {
-#if !(defined(__GNUC__) && (__GNUC__ == 4 && __GNUC_MINOR__ == 8))
+#if defined(__GNUC__) && !(__GNUC__ == 4 && __GNUC_MINOR__ == 8)
```

Comment:
Rather than repeat this mysterious-looking conditional a lot of places, can we define a more descriptive GNUC_BROKEN_ENUM_WARNING kind of thing and use it? Similarly for the CLANG ones?